src/coreclr/src/jit/morph.cpp

```diff
@@ -13263,13 +13276,6 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
 
                 if (isZeroOffset)
                 {
-                    // The "op1" node might already be annotated with a zero-offset field sequence.
-                    FieldSeqNode* existingZeroOffsetFldSeq = nullptr;
-                    if (GetZeroOffsetFieldMap()->Lookup(op1, &existingZeroOffsetFldSeq))
-                    {
-                        // Append the zero field sequences
-                        zeroFieldSeq = GetFieldSeqStore()->Append(existingZeroOffsetFldSeq, zeroFieldSeq);
-                    }
                     // Transfer the annotation to the new GT_ADDR node.
                     fgAddFieldSeqForZeroOffset(op1, zeroFieldSeq);
```

Comment:
Should there be an assert here that `zeroFieldSeq != nullptr` or is that self-evident from the fact that `isZeroOffset` is true?

Reply:
We could add it, but it looks self-evident.
Both values are initialized a few lines above with:
`bool          isZeroOffset = GetZeroOffsetFieldMap()->Lookup(tree, &zeroFieldSeq);`
so `isZeroOffset == true`  gurantees `zeroFieldSeq != nullptr`.

The problem with that code was that the condition was written before `fgAddFieldSeqForZeroOffset` was introduced. When `fgAddFieldSeqForZeroOffset` was created that logic was encapsulated there, but the condition was not deleted.

Reply:
Thanks for the clarification